build/release/release.sh

```diff
@@ -121,6 +121,9 @@ upload_nexus_staging() {
 
   echo "Deploying celeborn-client-mr-shaded_2.12"
   ${PROJECT_DIR}/build/sbt -Pmr "clean;celeborn-client-mr-shaded/publishSigned"
+
+  echo "Deploying celeborn-openapi-client_2.12"
+  ${PROJECT_DIR}/build/mvn clean deploy -DskipTests -pl openapi/openapi-client -s "${PROJECT_DIR}/build/release/asf-settings.xml"
```

Comment:
Prefer to use mvn to deploy due the maven plugin leverage higher openapi-generator version.

Reply:
And for openapi-generator, it will fail during `doc` phase.
<img width="997" alt="image" src="https://github.com/apache/celeborn/assets/6757692/7ed9ec32-b9c4-4636-a6dc-e2046148bd3c">
I also tried to use `sbt-openapi-generato` 7.7.0(the latest one) with JDK17.
The similar issue:
<img width="955" alt="image" src="https://github.com/apache/celeborn/assets/6757692/a23d6abd-8735-48bb-bf95-414136d47bec">

I also tried that in `unitycatalog` repo, simliar issue.
<img width="1012" alt="image" src="https://github.com/apache/celeborn/assets/6757692/efd5c24c-8c9a-4abb-8ef5-15c768987417">

So, two workaround:

1. using maven
2. using sbt and skip the `doc` phase.


Prefer to use mvn to deploy due the maven plugin openapi-generator version is higher

Reply:
I lean toward option 2, because it's a burden to maintain two building systems in long term, let's use sbt as much as possible.

Reply:
done,
verified the command:

```
(base) ➜  celeborn git:(deploy_openapi_client) ✗ build/sbt "clean;celeborn-openapi-client/publishSigned"
Using /Users/fwang12/.sdkman/candidates/java/current as default JAVA_HOME.
Note, this will be overridden by -java-home if it is set.
[info] welcome to sbt 1.9.4 (Azul Systems, Inc. Java 1.8.0_392)
[info] loading settings for project celeborn-build from plugins.sbt ...
[info] loading project definition from /Users/fwang12/todo/celeborn/project
[info] compiling 1 Scala source to /Users/fwang12/todo/celeborn/project/target/scala-2.12/sbt-1.0/classes ...
[info] loading settings for project celeborn from version.sbt ...
[info] resolving key references (11565 settings) ...
[info] set current project to celeborn (in build file:/Users/fwang12/todo/celeborn/)
[success] Total time: 0 s, completed Jul 11, 2024 9:01:07 AM
[info] generate META-INF/LICENSE for project celeborn-openapi-client
[info] generate META-INF/NOTICE for project celeborn-openapi-client
Cleaning up openapi client output directory: openapi/openapi-client/target/generated-sources/java
[info] Wrote /Users/fwang12/todo/celeborn/openapi/openapi-client/target/scala-2.12/celeborn-openapi-client_2.12-0.6.0-SNAPSHOT.pom
[info] generate META-INF/LICENSE for project celeborn-openapi-worker-client
[info] generate META-INF/LICENSE for project celeborn-openapi-master-client
[info] generate META-INF/NOTICE for project celeborn-openapi-worker-client
[info] generate META-INF/NOTICE for project celeborn-openapi-master-client
################################################################################
################################################################################
# Thanks for using OpenAPI Generator.                                          #
# Thanks for using OpenAPI Generator.                                          #
# Please consider donation to help us maintain this project 🙏                 #
# https://opencollective.com/openapi_generator/donate                          #
################################################################################
# Please consider donation to help us maintain this project 🙏                 #
# https://opencollective.com/openapi_generator/donate                          #
################################################################################
[info] Successfully generated code to /Users/fwang12/todo/celeborn/openapi/openapi-client/target/generated-sources/java
[info] Successfully generated code to /Users/fwang12/todo/celeborn/openapi/openapi-client/target/generated-sources/java
[info] compiling 88 Java sources to /Users/fwang12/todo/celeborn/openapi/openapi-client/target/scala-2.12/classes ...
[info] /Users/fwang12/todo/celeborn/openapi/openapi-client/target/generated-sources/java/src/main/java/org/apache/celeborn/rest/v1/master/invoker/JSON.java: Some input files use or override a deprecated API.
[info] /Users/fwang12/todo/celeborn/openapi/openapi-client/target/generated-sources/java/src/main/java/org/apache/celeborn/rest/v1/master/invoker/JSON.java: Recompile with -Xlint:deprecation for details.
[warn] Could not determine source for class org.apache.celeborn.rest.v1.worker.invoker.model.AbstractOpenApiSchema
[warn] Could not determine source for class org.apache.celeborn.rest.v1.master.invoker.model.AbstractOpenApiSchema
[warn] versionScheme setting is empty; set `ThisBuild / versionScheme := Some("early-semver")`, `Some("semver-spec")` or `Some("pvp")`
[warn] so tooling can use it for eviction errors etc - https://www.scala-sbt.org/1.x/docs/Publishing.html
[info] gpg: signing failed: Inappropriate ioctl for device
[info] gpg: signing failed: Inappropriate ioctl for device
[error] java.lang.RuntimeException: Failure running 'gpg --detach-sign --armor --use-agent --output /Users/fwang12/todo/celeborn/openapi/openapi-client/target/scala-2.12/celeborn-openapi-client_2.12-0.6.0-SNAPSHOT.pom.asc /Users/fwang12/todo/celeborn/openapi/openapi-client/target/scala-2.12/celeborn-openapi-client_2.12-0.6.0-SNAPSHOT.pom'.  Exit code: 2
[error]         at scala.sys.package$.error(package.scala:30)
[error]         at com.jsuereth.sbtpgp.CommandLineGpgSigner.sign(PgpSigner.scala:82)
[error]         at com.jsuereth.sbtpgp.PgpSettings$.$anonfun$signingSettings$2(PgpSettings.scala:151)
[error]         at scala.collection.TraversableLike.$anonfun$flatMap$1(TraversableLike.scala:293)
[error]         at scala.collection.immutable.Map$Map4.foreach(Map.scala:493)
[error]         at scala.collection.TraversableLike.flatMap(TraversableLike.scala:293)
[error]         at scala.collection.TraversableLike.flatMap$(TraversableLike.scala:290)
[error]         at scala.collection.AbstractTraversable.flatMap(Traversable.scala:108)
[error]         at com.jsuereth.sbtpgp.PgpSettings$.$anonfun$signingSettings$1(PgpSettings.scala:146)
[error]         at scala.Function1.$anonfun$compose$1(Function1.scala:49)
[error]         at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:63)
[error]         at sbt.std.Transform$$anon$4.work(Transform.scala:69)
[error]         at sbt.Execute.$anonfun$submit$2(Execute.scala:283)
[error]         at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:24)
[error]         at sbt.Execute.work(Execute.scala:292)
[error]         at sbt.Execute.$anonfun$submit$1(Execute.scala:283)
[error]         at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:265)
[error]         at sbt.CompletionService$$anon$2.call(CompletionService.scala:65)
[error]         at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[error]         at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
[error]         at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[error]         at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[error]         at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[error]         at java.lang.Thread.run(Thread.java:750)
[error] (celeborn-openapi-client / signedArtifacts) Failure running 'gpg --detach-sign --armor --use-agent --output /Users/fwang12/todo/celeborn/openapi/openapi-client/target/scala-2.12/celeborn-openapi-client_2.12-0.6.0-SNAPSHOT.pom.asc /Users/fwang12/todo/celeborn/openapi/openapi-client/target/scala-2.12/celeborn-openapi-client_2.12-0.6.0-SNAPSHOT.pom'.  Exit code: 2
[error] Total time: 3 s, completed Jul 11, 2024 9:01:10 AM
```